dynamic_instrumentation/src/instrument.rs

```diff
@@ -485,29 +527,29 @@ impl<'tcx> Visitor<'tcx> for CollectInstrumentationPoints<'_, 'tcx> {
                     .dest(&dest)
                     .add_to(self);
             }
-            Rvalue::AddressOf(_, p) => {
-                // Instrument which local's address is taken
-                self.loc(location, location.successor_within_block(), addr_local_fn)
-                    .arg_var(dest)
-                    .arg_index_of(p.local)
-                    .source(p)
-                    .dest(&dest)
-                    .instrumentation_priority(InstrumentationPriority::Early)
-                    .add_to(self);
-            }
             Rvalue::Use(Operand::Copy(p) | Operand::Move(p)) if p.is_indirect() => {
                 // We're dereferencing something, the result of which is a reference or pointer
                 self.loc(location, location.successor_within_block(), load_value_fn)
                     .arg_var(dest)
                     .dest(&dest)
                     .add_to(self);
             }
-            Rvalue::Use(Operand::Constant(..)) => {
+            Rvalue::Use(op @ Operand::Constant(..)) => {
+                let const_ty = match op_ty(op).kind() {
+                    TyKind::RawPtr(TypeAndMut { ty, .. }) => *ty,
+                    TyKind::Ref(_, ty, _) => *ty,
+                    ty @ _ => unimplemented!("Unsupported constant ty {:?}", ty)
+                };
```

Comment:
I guess there's logic somewhere that prevents this from being called on non-pointers?  Otherwise `x = 0` would hit this `unimplemented!` case.

Reply:
Yes, line 475.
```
            _ if !is_region_or_unsafe_ptr(value_ty) => {}
```